pkg/registry/core/service/rest_test.go

```diff
@@ -518,11 +532,28 @@ func TestServiceRegistryUpdateExternalService(t *testing.T) {
 	if _, _, err := storage.Update(ctx, svc3.Name, rest.DefaultUpdatedObjectInfo(svc3, api.Scheme)); err != nil {
 		t.Fatalf("Unexpected error: %v", err)
 	}
+
+	srv, err := registry.GetService(ctx, svc3.Name, &metav1.GetOptions{})
```

Comment:
This block looks like it gets used over and over; could we extract it out into its own function and just call that from each testcase?

Also, perhaps cleanup should be done from a defer instead?  Otherwise t.Errorf() will continue to the next testcase and have the same problem.  So I guess two solutions, either push the cleanup to a defer, or change all the t.Errorf() to t.Fatalf() so that the test stops when the error happens.